public/app/features/auth-config/GitHubConfigPage.tsx

```diff
@@ -0,0 +1,176 @@
+  loadSettings,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+export type Props = ConnectedProps<typeof connector>;
+
+/**
+ * Separate the Page logic from the Content logic for easier testing.
+ */
+export const GitHubConfigPage = ({ settings, loadSettings, isLoading }: Props) => {
+  useEffect(() => {
+    if (!settings) {
+      loadSettings();
+    }
+  }, [settings, loadSettings]);
+  return (
+    <Page navId="authentication" pageNav={pageNav}>
+      <GitHubConfig settings={settings} isLoading={isLoading} />
+    </Page>
+  );
+};
+
+export default connector(GitHubConfigPage);
+
+interface GitHubConfigProps {
+  settings?: SSOProvider;
+  isLoading?: boolean;
+}
+
+export const GitHubConfig = ({ settings }: GitHubConfigProps) => {
+  const [isSaving, setIsSaving] = useState(false);
+  const handleSubmit = async (data: ProviderData) => {
+    setIsSaving(true);
+    const requestData = dtoToData<ProviderData>(data);
+    try {
+      await getBackendSrv().put('/api/v1/sso-settings/github', requestData);
+      appEvents.publish({
+        type: AppEvents.alertSuccess.name,
+        payload: ['Settings saved'],
+      });
+    } catch (error) {
+      let message = '';
+      if (isFetchError(error)) {
+        message = error.data.message;
+      } else if (error instanceof Error) {
+        message = error.message;
+      }
+      appEvents.publish({
+        type: AppEvents.alertError.name,
+        payload: [message],
+      });
+    } finally {
+      setIsSaving(false);
+    }
+  };
+
+  return (
+    <Page.Contents>
+      <Stack grow={1} direction={'column'}>
+        <Form onSubmit={handleSubmit} defaultValues={dataToDTO(settings)}>
+          {({ register, errors, control, setValue, watch, formState }) => {
+            const teamIdOptions = watch('teamIds');
+            const orgOptions = watch('allowedOrganizations');
+            return (
+              <>
+                <FormPrompt confirmRedirect={formState.isDirty} onDiscard={() => {}} />
+                <Field label="Enabled">
+                  <Switch {...register('enabled')} id="enabled" label={'Enabled'} />
+                </Field>
+                <Field label="Client ID" required invalid={!!errors.clientId} error="This field is required">
+                  <Input {...register('clientId', { required: true })} type="text" id="clientId" />
+                </Field>
+                <Field label="Client secret" required invalid={!!errors.clientSecret} error="This field is required">
+                  <Input {...register('clientSecret', { required: true })} type="text" id="clientSecret" />
+                </Field>
+                <Field label="Team IDs" htmlFor={'teamIds'}>
+                  <InputControl
+                    name={'teamIds'}
+                    control={control}
+                    render={({ field: { ref, onChange, ...fieldProps } }) => {
+                      return (
+                        <Select
```

Comment:
Validate that the `teamIds` are numbers.

Reply:
Should they be numbers? Currently they are sent as a string of comma-separated values.

Reply:
> Should they be numbers? 

Yes, please validate that the entries in the select are numbers and send it in the `Extra` as a comma-separated string.